### ISSUE TITLE
Write Windows Instructions for R6 Dev Environment Using Docker

### ISSUE THREAD

**Issue by mcwhitaker** [avalonmediasystem/avalon#1751]
### So that
I can do development work on R6 on a Windows machine.

### Parent Issue
#1704

### Tasks
- [x] Documentation page with instructions available on the wiki
https://github.com/avalonmediasystem/avalon/wiki/Configuring-a-Windows-Development-Environment-with-Docker
- [x] Inaccurate or deprecated documentation is removed from the wiki
- [ ] Document whether quickstart works on Windows or only in Linux VM.
- [ ] Documentation is tested and validated by a developer other than the one who created the instructions

**Comment by mcwhitaker** [comment]
pages such as this one: https://wiki.dlib.indiana.edu/display/VarVideo/Avalon+6+Development+Setup

should become obsolete with this work.

**Comment by davidschober** [comment]
Can we augment the docker instructions?

**Comment by carrickr** [comment]
I really am opposed to this in general.  Project Hydra has never really supported Dev Work on Windows, so we'd be taking this on for ourselves.  We'd be supporting three dev environments with a relatively small team and that seems like a poor allocation of resources.

We'd have issues with the fact that our tools themselves do not play nice with Windows.  Ruby development on Windows is somewhere between painful and completely unsupported.  Typically when do you Ruby development on Windows 10, Step 1 is to install the Linux subsystem.  At which point just go Linux.

Docker also does not play well with Windows.

So we're in a situation where we're either totally dependent on Win10's Linux subsystem or we boot up a Linux VM, load up docker inside of the VM and then deal with the fact that we now have docker containers inside a VM and are paying performance penalties for virtualizing inside a virtual container and battling it out with Virtual Box's networking interfaces.

Basically our paths would be:

- Install Linux mode on a Win10 box and have a developer document the steps involved in getting that installed, then installing Ruby, docker, etc onto that Linux mode (and deal with any networking problems that may arise).  The developer in question also probably needs to learn about Linux mode.  Then hope MS never breaks it.
- Do the whole docker container inside a VM thing.

Both of which seem like a fair amount of developer time for little return.

**Comment by carrickr** [comment]
Per conversations on slack, people know of Villanova using Windows for development, but generally folks recommend against it.

**Comment by phuongdh** [comment]
I've seen some weird Docker file permissions on Windows, but the biggest pain is you'd have to enable HyperV and VirtualBox won't work as a result.

I think this story is more like "Get Docker-based dev setup in my Linux VM" now. There's a performance penalty but it's better than not having transcoding and streaming.

**Comment by cjcolvar** [comment]
As much as it isn't the prettiest, we do have devs who have windows machines and need a dev environment whatever that looks like.  So at the least we need internal instructions for onboarding a new dev with a windows machine which could be as dead simple as install a linux VM and use the linux instructions despite the performance penalty.

**Comment by davidschober** [comment]
I think this can be low priority.

**Comment by mcwhitaker** [comment]
@davidschober While this is not done, only mac users do the support of R6.? At IU, only Chris will be able to do that.

**Comment by cjcolvar** [comment]
I don't think that is entirely true, but it will consume more time and be unknown if dev environments on Windows machines are complete and good for testing without doing this.

**Comment by davidschober** [comment]
We will have dev instructions for both linux and mac boxes. We can simply advise that if working on a windows machine, install a linux vm then run the docker instructions. I agree with Chris C. and Carrick's general assessment.

**Comment by mcwhitaker** [comment]
@davidschober The description of the story does not dictate any particular approach. I am not disagreeing with the approach of VM plus linux, which I actually verbally agreed to during backlog refinement. All I am saying is that we need the check boxes of the Done Looks Like of this story completed -- and I do not agree that that is low priority.

**Comment by phuongdh** [comment]
@davidschober Pretty sure this can be abandoned since we now have an easier way https://github.com/avalonmediasystem/avalon#quickstart-development-with-docker-experimental

**Comment by davidschober** [comment]
Cool!
